Client/game_sa/CPoolsSA.h

```diff
@@ -54,8 +62,76 @@ class CPoolSAInterface
         m_bOwnsAllocations = false;
     }
 
+    uint GetFreeSlot()
+    {
+        bool bLooped = false;
+        uint index = m_nFirstFree + 1;
+
+        while (true)
+        {
+            if (index >= m_nSize)
+            {
+                if (bLooped)
+                    return -1;
+
+                index = 0;
+                bLooped = true;
+            }
+
+            if (m_byteMap[index].bEmpty)
+            {
+                m_nFirstFree = index;
+                return index;
+            }
+            index++;
+        }
+
+        return -1;
+    };
+
+    B* Allocate()
+    {
+        uint index = GetFreeSlot();
+        if (index == -1)
+            return nullptr;
+
+        return AllocateAt(index);
```

Comment:
Improved version of the original code.
```suggestion
        m_nFirstFree++;          // Continue after the last allocated slot
        const auto sz = m_nSize; // Storing size to avoid reloads from memory - should help out the optimizer
        for (auto i = 0u; i < sz; i++) {
            const auto slot = (m_nFirstFree + i) % sz;
            const auto e    = &m_byteMap[slot];
            if (!e->bEmpty) {
                continue;
            }
            e->bEmpty = false;
            e->nId++;
            return &m_pObjects[slot];
        }
        return nullptr;
```

Reply:
I suggest 
```c++
m_nFirstFree = slot;
```
after
```c++
 `e->nId++;` 
```

Reply:
I suggest
```c++
m_nFirstFree = slot;
```
This helps skip original allocated items

Reply:
Ah shit I forgot to add that.
Thanks!